packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,270 @@
+                    }
+
+                    try {
+                        const query = documentType.getQuery;
+
+                        if (query) {
+                            const { data } = await client.query<GQLPageQuery, GQLPageQueryVariables>({
+                                query,
+                                variables: {
+                                    id: page.id,
+                                },
+                                context: LocalErrorScopeApolloContext,
+                            });
+
+                            if (data?.page?.document) {
+                                const extractedContent = documentType.extractTextContents?.(data.page.document);
+                                textContents.push(page.name, page.slug, ...(extractedContent ?? []));
+                            }
+                        }
+                    } catch (e) {
+                        throw new Error(`Error while fetching page`);
+                    }
+                }),
+            );
+
+            const pageText = {
+                textContents: Array.from(new Set(textContents.filter((content) => content && content !== ""))),
+            };
+
+            return JSON.stringify(pageText);
+        },
+        [client, documentTypes],
+    );
+
+    const importContents = React.useCallback(
+        async (pages: GQLPageTreePageFragment[], content: ParsedContents): Promise<void> => {
+            await Promise.all(
+                pages.map(async (page) => {
+                    const documentType = documentTypes[page.documentType];
+
+                    if (!documentType) {
+                        throw new Error(`Unknown document type "${documentType}"`);
+                    }
+
+                    try {
+                        const query = documentType.getQuery;
+
+                        if (query) {
+                            const { data } = await client.query<GQLPageQuery, GQLPageQueryVariables>({
+                                query,
+                                variables: {
+                                    id: page.id,
+                                },
+                                context: LocalErrorScopeApolloContext,
+                            });
+
+                            if (data?.page?.document) {
+                                const documentWithUpdateContents = documentType.replaceTextContents?.(data.page.document, content.contents);
+
+                                if (documentType.updateMutation) {
+                                    await client.mutate<GQLUpdatePageMutation, GQLUpdatePageMutationVariables>({
+                                        mutation: documentType.updateMutation,
+                                        variables: {
+                                            pageId: data.page.document.id,
+                                            input: { content: documentWithUpdateContents?.content, seo: documentWithUpdateContents?.seo },
```

Comment:
This input is application-specific here you need do call `documentType.inputToOutput`